compact.go

```diff
@@ -914,93 +947,109 @@ func (c *compactionSeriesSet) Err() error {
 	return c.p.Err()
 }
 
-func (c *compactionSeriesSet) At() (labels.Labels, []chunks.Meta, Intervals) {
-	return c.l, c.c, c.intervals
+func (c *compactionSeriesSet) At() (uint64, labels.Labels, []chunks.Meta, Intervals) {
+	return c.p.At(), c.l, c.c, c.intervals
 }
 
 type compactionMerger struct {
-	a, b ChunkSeriesSet
+	sets  []ChunkSeriesSet
+	first bool
+	oks   []bool
+	// TODO(enhancement): If we can know number of series in each block, we can
```

Comment:
Are the series ids consecutive though?

Reply:
I added `NumSeries()` for `BlockReader` and tried to use slice instead of map. Though I pre-allocate using `NumSeries()`, I was taking care if the series ref went beyond the slice and append appropriately. Though this was faster than upstream and B/op was less than upstream, B/op was still more than using map. 
One strange thing that I found out: The series ref went beyond the number of series. I saw series ref going upwards of 4M though number of series are 1M (this could be the reason why B/op was more than using map). @gouthamve this shouldn't happen right?

Reply:
That can happen.

Reply:
I thought series ref spanned from `0` to `numSeries-1`.

Reply:
Not sure if I was using the right term, but postings should span only till `numSeries-1` according to this
https://github.com/prometheus/tsdb/blob/6ab483071a5d4dce43634f6eef2dce67d1f4740f/compact.go#L819

Reply:
I confused between id and reference (the offset). I will try id and see if it helps.